src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1678,6 +1678,12 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
 
     double EclipseGrid::getCellDepth(size_t globalIndex) const {
         assertGlobalIndex( globalIndex );
+
+        auto it = this->m_aquifer_cell_depths.find(globalIndex);
+        return it != this->m_aquifer_cell_depths.end() ? it->second : computeCellDepth(globalIndex);
+    }
+
+    double EclipseGrid::computeCellDepth(size_t globalIndex) const {
```

Comment:
I suggest renaming this `computeCellGeometricDepth()` or similar, to make clearer the difference between this and `getCellDepth()`.